hack/release-helper.sh

```diff
@@ -0,0 +1,257 @@
+            echo "Error: Missing ARGS"
+            usage_and_exit
+        fi
+    fi
+}
+
+
+#
+# Make sure that the pre-reqs container ships the same version of the nydus
+# snapshotter that we use in the kata ci. Update the Makefile if needed.
+#
+function check_nydus() {
+    echo "Checking that the operator's nydus version matches the one used in kata"
+
+    kata_versions="https://raw.githubusercontent.com/kata-containers/kata-containers/main/versions.yaml"
+    kata_versions_dest="versions.yaml.$$"
+
+    # pull the versions.yaml that kata uses
+    wget --quiet "${kata_versions}" -O "${kata_versions_dest}"
+
+    # multi-line grep to find the nydus-snapshotter version kata's version.yaml
+    # then awk/tail/tr and string manipulation to sand it down.
+    nydus_version_kata="$(grep -zoP "nydus-snapshotter.*\n.*description.*\n.*url.*\n.*version.*" ${kata_versions_dest} | awk '{print $2}' | tail -n1 | tr -d '\0')"
+    nydus_version_kata=${nydus_version_kata//\"} # strip double quotes
+
+    # delete temporary kata versions.yaml that we downloaded
+    rm "${kata_versions_dest}"
+
+    # grab the nydus version in the operator
+    prereqs_makefile="install/pre-install-payload/Makefile"
+    nydus_version_prereqs="$(grep 'NYDUS_SNAPSHOTTER_VERSION = ' $prereqs_makefile | awk '{ print $3}')"
+
+    # check for match and update if needed
+    if [ "${nydus_version_kata}" == "${nydus_version_prereqs}" ]; then
+        echo "-> OK, no changes made: nydus-snapshotter already up-to-date" \
+             "(version: $nydus_version_prereqs)"
+    else
+        echo "-> Found mismatch: updating nydus-snapshotter from" \
+             "$nydus_version_prereqs to $nydus_version_kata"
+        sed -i "s/NYDUS_SNAPSHOTTER_VERSION = .*/NYDUS_SNAPSHOTTER_VERSION = $nydus_version_kata/" "${prereqs_makefile}"
+        echo "-> Changes made to ${prereqs_makefile}."
+        echo "   Please verify and open PR as needed."
+    fi
+}
+
+
+#
+# Update the operator tag for the new release.
+#
+function update_operator() {
+    operator_yaml="config/release/kustomization.yaml"
+    echo "Updating operator version to ${operator_tag}"
+    echo "-> Updating ${operator_yaml}"
+    sed -i 's/newTag:.*/newTag: '${operator_tag}'/' ${operator_yaml}
+}
+
+
+#
+# Update the CRDs to point to the most recent version of the pre-reqs payload.
+#
+function update_prereqs() {
+    # grab the hash for the latest commit on the pre-install-payload folder
+    # in github (which requires figuring out the hash of the merge)
+    branch=HEAD
+    commit="$(git log -1 --pretty=format:"%H" \
+              install/pre-install-payload/)"
+    prereqs_latest_hash=$((git rev-list $commit..$branch --ancestry-path | cat -n; git rev-list $commit..$branch --first-parent | cat -n) | sort -k2 -s | uniq -f1 -d | sort -n | tail -1 | cut -f2)
+
+    echo "Updating pre-reqs payload to $prereqs_latest_hash"
+
+    prereqs_yaml=(
```

Comment:
Hi @portersrc !

The list of files in `prereqs_yaml` and `kata_yaml` can get outdated. How can we ensure it's reviewed before each release?

I thought in either:
 * move those variables to a separated file which should be source'ed by this script. Then on release guide we mention that file should be reviewed before the release
 * instead of a separated file, just make them into a global variable at the top of the script. Likewise document on the release guide

Reply:
In the long run, the first option looks slightly easier for someone doing the release, so let's go with that. This is a great point, and burying those lists in the script was bad.

Reply:
added the note to the release template [here](https://github.com/confidential-containers/confidential-containers/pull/246).